cli/src/utils.rs

```diff
@@ -4,18 +4,22 @@
 
 use clap::Args;
 use color_eyre::eyre;
+
 use std::{fs, io::Read, path::PathBuf};
 use subxt_codegen::utils::{MetadataVersion, Uri};
 
+pub mod type_description;
+pub mod type_example;
+
 /// The source of the metadata.
 #[derive(Debug, Args)]
 pub struct FileOrUrl {
     /// The url of the substrate node to query for metadata for codegen.
     #[clap(long, value_parser)]
-    url: Option<Uri>,
+    pub(crate) url: Option<Uri>,
     /// The path to the encoded metadata file.
     #[clap(long, value_parser)]
-    file: Option<PathBuf>,
+    pub(crate) file: Option<PathBuf>,
```

Comment:
Nit: probably fine just being `pub`, ie no `(crate)` (this is a binary anyway atm and even if we make a bunch of it a lib for testing nobody should depend on it.